### DISCUSSION TITLE
how to get ALE highliht the entire error intead of the first character?

### DISCUSSION THREAD

**Discussion by hwijeen** [w0rp/ale#4558] [Q&A - Ask for help with problems]
Hi, I'm enjoying configuring ALE, but I can't figure out how to get ALE to highlight the entire selection (e.g. variable).
My configuration only highlights the first character of the problem. How do I fix this?

<img width="856" alt="image" src="https://github.com/dense-analysis/ale/assets/29157715/c245c74e-8d10-4b8c-911b-de18d9603269">

I suspect this has to do with vim's highlight setting but I can't figure why it's not working.

<img width="471" alt="image" src="https://github.com/dense-analysis/ale/assets/29157715/6a732d5a-3b92-4612-895a-9577f91e255b">

**Comment by hsanson** [comment]
Cannot reproduce this.  Please try running vim with a minimal configuration that loads only ALE and then keep adding other plugins to see if any of those is causing the issue.

**Reply by hwijeen** [reply]
Thank you for the answer. Is there a way to inspect the variable that ALE keeps in ordoer to mark the string to be highlighted?

**Reply by w0rp** [reply]
Which language and for which linter do you see this behaviour? Probably the linter needs updating to it specifies the end column for the warning so ALE knows which characters to highlight. I suspect ALE is only being told to highlight the first character.

**Reply by hwijeen** [reply]
Thank you for the answer! I am using python. Could you point me to the relevant variables that control it? I tried tweaking highlight options but it didn't work.

**Reply by hsanson** [reply]
This depends on the linter implementation you are using. If the linter provides end_col then ALE can use it, if it does not provide that information then there is no way for ALE to highlight the whole error.

You can see what linters you are using using `:ALEInfo`.

**Reply by hwijeen** [reply]
I am using `let g:ale_linters = {'python': ['pylint', 'pylsp']}`, could you share which ones you are using?
